eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -26,6 +26,10 @@
     "AdditionalTestArguments": [
         "/p:UseProjectReferenceToAzureClients=false",
         "/p:UseProjectReferenceToAzureClients=true"
+    ],
+    "BuildConfiguration": [
+        "Debug",
+        "Release"
```

Comment:
Shouldn't this line be removed, because you are doing the `MatrixReplace` in the calling yaml? Otherwise I think you'll end up with two identical jobs that both run for `Release`, since the `BuildConfiguration` array will be `["Release", "Release"]`.

Reply:
The point of the replace it to have all `internal` test pipelines be `Release`.

So we go from
```
Build Test PackageRef_macOS1015_net60_Debug 
Build Test PackageRef_ubuntu2004_netcoreapp31_Debug 
Build Test PackageRef_windows2019_net461_Release 
Build Test ProjectRef_macOS1015_net60_Debug 
Build Test ProjectRef_ubuntu2004_netcoreapp31_Debug 
Build Test ProjectRef_windows2019_net461_Release 
Build Test windows2019_netcoreapp31_Coverage_Record  
```
to
```
Build Test PackageRef_macOS1015_net60_Release 
Build Test PackageRef_ubuntu2004_netcoreapp31_Release 
Build Test PackageRef_windows2019_net461_Release 
Build Test ProjectRef_macOS1015_net60_Release 
Build Test ProjectRef_ubuntu2004_netcoreapp31_Release 
Build Test ProjectRef_windows2019_net461_Release 
Build Test windows2019_netcoreapp31_Coverage_Record  
```